backend/src/controllers/items-controller.ts

```diff
@@ -0,0 +1,136 @@
+        itemOwnerId: itemOwnerId
+    });
+    //Creates a transaction to create inventory-items
+    //Checks if there's missing sizes
+    const transaction = await mongoose.startSession();
+    transaction.startTransaction();
+    try {
+        const savedItem = await item.save();
+        const sizeList = ['XXS', 'XS', 'S', 'M', 'L', 'XL', 'XXL'];
+        for (const size of sizeList) {
+            const invItem = new InventoryItem({
+                size: size,
+                item: savedItem._id
+            });
+            await invItem.save({session: transaction});
+        }
+        await transaction.commitTransaction();
+        await transaction.endSession();
+        return res.status(201).json({item});
+    } catch (error: any) {
+        await transaction.abortTransaction();
+        await transaction.endSession();
+        return res.status(500).json({message: 'Server error', error: error.message});
+    }
+}
+
+
+//Retrieves a list of all items
+export const retrieveItems = async (req: Request, res: Response) => {
+    try {
+        const items = await Item.find();
+        return res.status(200).json(items);
+    } catch (error: any) {
+        return res.status(500).json({ message: 'Server error', error: error.message });
+    }
+}
+
+//Retrieves details of a specific item
+export const getSpecificItem = async (req: Request, res: Response) => {
+    try {
+        const itemID: string = req.params._id;
+        const desiredItem = await Item.findOne({ _id: itemID });
+        if (desiredItem){
+            return res.status(200).json(desiredItem);
+        }
+        else{
+            return res.status(404).json({ error: "No item with the provided ID exists"});
+        }
+    } catch (error: any){
+        return res.status(500).json({ message: 'Server error', error: error.message });
+    }
+}
+
+//Updates a specific item
+export const updateItem = async (req: Request, res: Response) => {
+    const { newName }: { newName: string }= req.body;
+    const { newDescription }: { newDescription: string }= req.body;
+    const { newPickupLocation }: { newPickupLocation: string }= req.body;
+    const { newPickupTime }: { newPickupTime: string }= req.body;
+    const itemID: string = req.params._id;
+
+    //Validation error
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        console.log("errors", errors);
+        return res.status(400).json({ errors: errors.array() });
+    }
+    //Updates item
+    try {
+        const result = await Item.updateOne({_id: new ObjectId(itemID)},
+            {$set: { name: newName, description: newDescription, pickupLocation: newPickupLocation, pickupTime: newPickupTime}});
+        if(result.matchedCount === 0){
+            return res.status(404).json({ error: 'The desired item was not found'});
+        }
+        const updatedItem = await Item.findOne({ _id: new ObjectId(itemID)});
+        return res.status(200).json(updatedItem);
```

Comment:
```suggestion
        return res.status(200).json({updatedItem});
```
Similar suggestion to above